bundles/io/org.eclipse.smarthome.io.transport.mqtt/src/main/java/org/eclipse/smarthome/io/transport/mqtt/reconnect/PeriodicReconnectStrategy.java

```diff
@@ -53,8 +53,12 @@ public PeriodicReconnectStrategy(int reconnectFrequency, int firstReconnectAfter
 
     @Override
     public void lostConnection() {
-        logger.info("Starting connection helper to periodically try restore connection to broker '{}'",
-                getBrokerConnection().getName());
+        if (scheduledTask != null && !scheduledTask.isDone()) {
```

Comment:
Isn't connection established and lost connection called as a callback and potential from different thread. Can we really omit read / write locks? Are the functions called by the same thread all the time (also isReconnecting)?

Reply:
We can go with your PR instead, but it was work to create all those changes. Would be nice if you do all the additional things that this PR does.

Reply:
I don't care which one is used. I don't want to replace your one if just forget that you already fixed the reconnect issue.
So, I am fine with using yours, but we should still consider to find an accurate fix.
I didn't check e.g. "isDone()"

Reply:
I have no time to change this PR in the next days. But I can give you push access to my repo. I have seen that synchronized the access. Might be a good idea to make this thread safe, yes.

Reply:
@davidgraeff I added the reconnect relevant part of your changes to #4160. Can you review it and give your approval or comments what should be changed?

If we get me access rights to this one, I will remove the changes of the reconnect part from this PR, so it contains the reloading configuration fix only.